src/dash/models/DashManifestModel.js

```diff
@@ -1332,6 +1336,30 @@ function DashManifestModel() {
         });
     }
 
+    /* ESSENTIAL PROPERTY DESCRIPTORS */
+
+    function getEssentialPropertiesForAdaptation(adaptation) {
+        const essentialProperties = {};
+
+        if (adaptation && adaptation.hasOwnProperty(DashConstants.ESSENTIAL_PROPERTY_ASARRAY)) {
+            for (const ep of adaptation.EssentialProperty_asArray) {
+                if (ep.hasOwnProperty(Constants.SCHEME_ID_URI)) {
+                    // N.B this will only work where there is a single EssentialProperty descriptor with this SchemeIdUri
+                    essentialProperties[ep[Constants.SCHEME_ID_URI]] = {...ep};
+                }
+            }
+        }
+        return essentialProperties;
+    }
+
+    function getEssentialPropertiesAsArrayForAdaptation(adaptation) {
+        if (!adaptation || !adaptation.hasOwnProperty(DashConstants.ESSENTIAL_PROPERTY_ASARRAY) || !adaptation.EssentialProperty_asArray.length) return [];
+        return adaptation.EssentialProperty_asArray.map( supp => {
```

Comment:
Please rename `supp` to `ep` or sth similar

Reply:
Changed to `ep`